vignettes/plotting_theme.Rmd

```diff
@@ -0,0 +1,309 @@
+---
+title: "Plotting Functions - Themes & Colors"
+output:
+  html_vignette:
+    number_sections: true
+    toc: true
+    toc_depth: 3
+    dfprint: kable
+
+vignette: >
+  %\VignetteIndexEntry{Plotting Functions - Themes & Colors}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  include = TRUE, 
+  eval = TRUE, 
+  echo = TRUE, 
+  warning = FALSE, 
+  message = FALSE, 
+  collapse = TRUE,
+  cache = FALSE, 
+  tidy = FALSE,
+  fig.height = 3.5, 
+  fig.width = 3.5,
+  fig.align = "left"
+  
+)
+
+## uncomment this when trying to knit vignette locally
+# library(ggpubr)
+# library(ggthemes)
+# library(ggpubr)
+# library(scales)
+# source("H:/GitHub/Rgemini/R/plot_theme.R")
+
+options(knitr.kable.NA = '')
+
+```
+
+
+*  *  *  *
+
+# Introduction
+
+This vignette introduces a standardized plotting theme ([*plot_theme()*](#plot_theme)) and color palettes ([*gemini_colors()*](#gemini_colors)) that users can apply to any ggplot figures they create in their analyses. The goal of these functions is to simplify the process of creating figures with a clean and consistent look. To create publication-ready figures, we recommend that users apply additional customization based on their needs (see section [Resources](#resources)). 
+
+
+*  *  *  *
+
+
+# Theme for `ggplots` {#plot_theme}
+
+`Rgemini::plot_theme()` adds a simple theme to any `ggplot` figures. The theme is based on `ggthemes::theme_foundation()`, with some additional tweaks, and users can easily adjust the theme as needed. By default, `plot_theme()` is applied to all other plotting functions in `Rgemini` (e.g., `plot_summary()` and `plot_over_time()`). 
+
+The following example illustrates the theme applied to a basic barplot. We also use the default color palette from `gemini_colors()` here for illustration purposes (see section [Color Palettes](#gemini_colors) for more details):
+
+```{r}
+library(Rgemini)
+library(ggplot2)
+
+# create example data
+my_data <- data.frame(group = c("A", "B", "C"), mean = c(5, 7, 4))
+
+# generate barplot
+my_plot <- ggplot(data = my_data, aes(x = group, y = mean, fill = group)) +
+  geom_bar(stat = "identity", fill = gemini_colors(1)[1:3]) +
+  scale_y_continuous(name = "Mean", expand = c(0, 0))
+
+# original ggplot theme
+my_plot + ggtitle("Original plot")
+```
+
+```{r}
```

Comment:
Hi, I would suggest here to add `library(ggthemes)`otherwise it would report error as `could not find function "theme_foundation"`

Reply:
Hi @youx-unity, note that calling `library(ggthemes)` is not required when the `Rgemini` package is installed (because `ggthemes` is already imported by the function `plot_theme()`). You can test this by installing `Rgemini` from this branch (or alternatively, you can load the required packages that are listed on lines 35-40 as a temporary solution but we won't need to add this here).